### PR TITLE
Feature/serial command improvements

### PR DESCRIPTION
**Pull request by LucHeart**
Implements #61 

In addition the parse able responses have been made more unique to parse by adding a dollar sign to $SYS$.

### CODE REVIEW

File: src/SerialInputHandler.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 #include <ArduinoJson.h>
 #include <Esp.h>
+#include "Logging.h"
```
Comment (hhvrc) [inline src/SerialInputHandler.cpp:8]: minor issue, but not big enough to require a change.

Try to order these in the order of:
Header file
Our headers
Library headers
Esp headers
C++ headers

File: src/SerialInputHandler.cpp [hunk 2]
```diff
@@ -123,52 +132,67 @@ void _handleFactoryResetCommand(char* arg, std::size_t argLength) {
 
 void _handleRmtpinCommand(char* arg, std::size_t argLength) {
   if (arg == nullptr || argLength <= 0) {
-    Serial.println("SYS|Error|Invalid argument");
+    // Get rmt pin
+    Serial.print("$SYS$|Response|RmtPin|");
```
Comment (hhvrc) [inline src/SerialInputHandler.cpp:136]: we should probably find a nicer way to print these responses 🤔

File: src/SerialInputHandler.cpp [hunk 3]
```diff
@@ -123,52 +132,67 @@ void _handleFactoryResetCommand(char* arg, std::size_t argLength) {
...
 }
 
 void _handleNetworksCommand(char* arg, std::size_t argLength) {
+
+
   if (arg == nullptr || argLength <= 0) {
-    Serial.println("SYS|Error|Invalid argument");
+    // Get networks
+    StaticJsonDocument<1024> outDoc;
+    JsonArray outNetworks = outDoc.to<JsonArray>();
```
Comment (hhvrc) [inline src/SerialInputHandler.cpp:169]: This works? 👀
Seems a bit cursed, but if it works, it works :D

**Review by hhvrc — APPROVED** [review]

File: src/SerialInputHandler.cpp (hunk 1 above)
Comment (hhvrc) [reply src/SerialInputHandler.cpp:8]: This is due to in some cases having the improper include order can mess up compilation in really weird ways that are incredibly hard to debug 😅

File: src/SerialInputHandler.cpp (hunk 2 above)
Comment (hhvrc) [reply src/SerialInputHandler.cpp:136]: Ah, saw the comment you had on the PR now, makes sense

File: src/SerialInputHandler.cpp (hunk 3 above)
Comment (LucHeart) [reply src/SerialInputHandler.cpp:169]: yup, dw about the weird arduino stuff lmao

File: src/SerialInputHandler.cpp (hunk 1 above)
Comment (LucHeart) [reply src/SerialInputHandler.cpp:8]: where do i put this header here then?

Comment (hhvrc) [reply src/SerialInputHandler.cpp:8]: Its not that important, but you would put it alongside "CommandHandler.h" and "Config.h"

Comment (LucHeart) [reply src/SerialInputHandler.cpp:8]: okay, but its not one of our headers? so you initial proposal doesnt apply here?

Comment (hhvrc) [reply src/SerialInputHandler.cpp:8]: Logging.h *is* one of our headers

**Review by hhvrc — APPROVED** [review]

**Pull request merged**